backend/src/main/java/com/htecgroup/skynest/model/dto/FolderDto.java

```diff
@@ -0,0 +1,32 @@
+package com.htecgroup.skynest.model.dto;
+
+import com.htecgroup.skynest.model.entity.BucketEntity;
+import com.htecgroup.skynest.model.entity.FolderEntity;
+import com.htecgroup.skynest.model.entity.UserEntity;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.With;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class FolderDto {
+  private UUID id;
+  private LocalDateTime createdOn;
+
+  private LocalDateTime modifiedOn;
+
+  @With private LocalDateTime deletedOn;
+
+  private UserEntity createdBy;
```

Comment:
Since you have a DTO, all attributes inside needs to be wrappers, primitives or other DTOs, you have Entity object here so it kinda messed the whole concept